### PR TITLE
Change variable name to avoid variable shadowing

### PR DESCRIPTION
**Pull request by JoseSanchez7**
**Problem:** When compiling with ARM, a warning is given that variable shadowing is possibly occurring. This was caused by the object type and a variable of that type having the same name (despite having different capitalization). While the compiler understands what's going on, it may lead to errors in understanding for others reading the code. 

**Solution:** Changing the variable name resolved the issue. 

**Links for more info:**

https://stackoverflow.com/questions/38533407/how-bad-is-redefining-shadowing-a-local-variable

https://stackoverflow.com/questions/15066925/same-name-but-with-different-case-variable-and-function-names-in-c

### CODE REVIEW

**Comment by vaavva** [comment]
Comment to hold the merge until the LTS release is complete

File: adapters/tlsio_mbedtls.c [hunk 1]
```diff
@@ -713,7 +713,7 @@ int tlsio_mbedtls_close(CONCRETE_IO_HANDLE tls_io, ON_IO_CLOSE_COMPLETE on_io_cl
     return result;
```
Comment (vaavva) [inline adapters/tlsio_mbedtls.c:713]: placeholder comment to hold the merge until the LTS release is complete

**Review by vaavva — APPROVED** [review]

File: adapters/tlsio_mbedtls.c (hunk 1 above)
Comment (vaavva) [inline adapters/tlsio_mbedtls.c:713]: placeholder comment to hold merge until LTS release is complete

**Review by vaavva — CHANGES_REQUESTED** [review]

File: adapters/tlsio_mbedtls.c [hunk 2]
```diff
@@ -713,7 +713,7 @@ int tlsio_mbedtls_close(CONCRETE_IO_HANDLE tls_io, ON_IO_CLOSE_COMPLETE on_io_cl
...
 }
 
-int tlsio_mbedtls_send(CONCRETE_IO_HANDLE tls_io, const void *buffer, size_t size, ON_SEND_COMPLETE on_send_complete, void *callback_context)
+int tlsio_mbedtls_send(CONCRETE_IO_HANDLE tls_io, const void *buffer, size_t size, ON_SEND_COMPLETE on_send_complete_obj, void *callback_context)
```
Comment (ericwol-msft) [inline adapters/tlsio_mbedtls.c:716]:
```suggestion
int tlsio_mbedtls_send(CONCRETE_IO_HANDLE tls_io, const void *buffer, size_t size, ON_SEND_COMPLETE on_send_complete_callback, void *callback_context)
```

File: adapters/tlsio_mbedtls.c [hunk 3]
```diff
@@ -732,7 +732,7 @@ int tlsio_mbedtls_send(CONCRETE_IO_HANDLE tls_io, const void *buffer, size_t siz
         }
         else
         {
-            tls_io_instance->send_complete_info.on_send_complete = on_send_complete;
+            tls_io_instance->send_complete_info.on_send_complete = on_send_complete_obj;
```
Comment (danewalton) [inline adapters/tlsio_mbedtls.c:735]: gotta change this guy too

Comment (JoseSanchez7) [reply adapters/tlsio_mbedtls.c]: Thanks @danewalton ! Any other changes you recommend?

**Review by vaavva — APPROVED** [review]

**Pull request merged**